www/widgets/DemoExampleMaker.js

```diff
@@ -291,6 +310,47 @@ class DemoExampleMaker extends Widget {
     this.$('[name="dem-url"]').style.display = 'block'
   }
 
+  _uploadJSON (ex) {
+    for (let step = this._data.steps.length; step >= 1; step--) {
+      this._updateStep(step, 'remove')
+    }
+    this._data = {
+      name: ex.name,
+      tags: ex.tags || null,
+      toc: ex.toc || true,
+      layout: ex.layout || 'dock-left',
+      key: Number(ex.key),
+      code: ex.code,
+      steps: ex.info ||
+      [{ title: 'getting started', focus: null, text: 'in this step...' }]
```

Comment:
similar to my note above, if we're opening an old example that doesn't have any steps, we shouldn't create one (we should let them create it) so the default value here should just be an empty array